### PR TITLE
Create contributing.md

### PR DESCRIPTION
**Pull request by pgroves**
Init contributing.md

### CODE REVIEW

**Comment by pgroves** [comment]
I added a 'contributing.md' file, which Github treats as a special file like README.md. It contains an overview and some notes on how we like to do Pull Requests.

I also copied some of our old documentation about Git and PRs to the wiki: https://github.com/ncsa/phyloflow/wiki. Those you can change directly if you want to add anything.

I'm also going to make everyone a Reviewer for this PR. Feel free to make comments or just click 'approve'. I just want everyone to see what it looks like.

File: contributing.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+## Pull Requests
+
+Once you have pushed a branch up to this repository, you can make a pull
+request through the github website. Some pointers: 
+1. Name your branch with
+your own initials, eg 'pg/readme_format' if your initials are pg and the  
+changes are about the formatting in the README. Do not worry too much about the name
+otherwise, just something short.
+2. Never commit directly to the `master` branch, always create a PR and click
+'merge' in the github website.
```
Comment (matthewberry) [inline contributing.md:25]: might want to clarify that we typically wait for approval before clicking merge

File: contributing.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+3. However, you can approve and merge your own pull requests in an emergency
+(or a tight deadline and it's 3am). Just make a PR so there is a record of it.
+You can't do anything that can't be undone.
```
Comment (matthewberry) [inline contributing.md:28]: some ambiguity on "can" here--possibility vs. permissibility. maybe "Anything you do can be undone if necessary."

File: contributing.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+4. **Try** to only have one functional change per PR. A functional change
+usually involves multiple files, so if there are multiple functional changes it takes a
+lot of documentation and mental effor for the reviewer to understand each
```
Comment (matthewberry) [inline contributing.md:31]: "effort"

File: contributing.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+changed file they see. EG, if you fix two bugs in one PR, and 10 files
+changed, the reviewer has a harder time understanding what each change is
+doing.  
+5. However, you will be forgiven if there is a lot in one PR (it won't
+be rejected). Just please be generous with explanations.
+6. If a PR is related to an issue, please make note of it in the PR
+description.  
+6. Please write a good overview of the changes as the `description`. If details
```
Comment (matthewberry) [inline contributing.md:39]: lines 37 and 39 are both "6.", which doesn't matter in a md viewer, but might be confusing in a text editor

there's also a hidden "6." in line 42

File: contributing.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+need explanation, it is easier to understand if they are put as comments on the
+diff. (I think you have to make the PR first, and then you can add comments
+inline).  6. You can make a PR even if it's not done. Just make a note in the
+description.
```
Comment (matthewberry) [inline contributing.md:43]: anything to say about process in this case--why make such a PR, and what happens to them?

**Review by matthewberry — APPROVED** [review]

**Review by melkebir — APPROVED** [review]
Sounds good. Thanks for sharing this.

**Review by ChuanyiZ — APPROVED** [review]
Thank you. I think this looks good.

**Pull request merged**